### PR TITLE
Stop enforcing jenkins plugin versions

### PR DESCRIPTION
**Pull request by tbrisker**
This will allow upgrading plugins from the UI

### CODE REVIEW

File: puppet/modules/jenkins_master/manifests/init.pp [hunk 1]
```diff
@@ -4,193 +4,193 @@
...
+    'rebuild'                            => {},
+    'redmine'                            => {},
+    'regression-report-plugin'           => {},
+    'ruby'                               => {},
+    'ruby-runtime'                       => {},
+    'rubyMetrics'                        => {},
+    'run-condition'                      => {},
+    'rvm'                                => {},
+    'saferestart'                        => {},
+    'schedule-failed-builds'             => {},
```
Comment (ekohl) [inline puppet/modules/jenkins_master/manifests/init.pp]: Can you drop this? installing the latest gives a 404

File: puppet/modules/jenkins_master/manifests/init.pp [hunk 2]
```diff
@@ -4,193 +4,193 @@
...
+    'postbuild-task'                     => {},
+    'project-stats-plugin'               => {},
+    'promoted-builds'                    => {},
+    'pubsub-light'                       => {},
+    'pxe'                                => {},
+    'rake'                               => {},
+    'rebuild'                            => {},
+    'redmine'                            => {},
+    'regression-report-plugin'           => {},
+    'ruby'                               => {},
```
Comment (ekohl) [inline puppet/modules/jenkins_master/manifests/init.pp]: Latest gives a 404

File: puppet/modules/jenkins_master/manifests/init.pp [hunk 3]
```diff
@@ -4,193 +4,193 @@
...
+    'pipeline-stage-step'                => {},
+    'pipeline-stage-tags-metadata'       => {},
+    'pipeline-stage-view'                => {},
+    'pipeline-utility-steps'             => {},
+    'plain-credentials'                  => {},
+    'postbuild-task'                     => {},
+    'project-stats-plugin'               => {},
+    'promoted-builds'                    => {},
+    'pubsub-light'                       => {},
+    'pxe'                                => {},
```
Comment (ekohl) [inline puppet/modules/jenkins_master/manifests/init.pp]: Latest gives a 404 so can be dropped.

File: puppet/modules/jenkins_master/manifests/init.pp [hunk 4]
```diff
@@ -4,193 +4,193 @@
...
+    'checkstyle'                         => {},
+    'cloudbees-bitbucket-branch-source'  => {},
+    'cloudbees-folder'                   => {},
+    'cobertura'                          => {},
+    'code-coverage-api'                  => {},
+    'command-launcher'                   => {},
+    'conditional-buildstep'              => {},
+    'config-file-provider'               => {},
+    'configurationslicing'               => {},
+    'copy-to-slave'                      => {},
```
Comment (ekohl) [inline puppet/modules/jenkins_master/manifests/init.pp]: Latest gives a 404 so can be dropped.

File: puppet/modules/jenkins_master/manifests/init.pp [hunk 5]
```diff
@@ -4,193 +4,193 @@
...
+    'blueocean-pipeline-api-impl'        => {},
+    'blueocean-pipeline-editor'          => {},
+    'blueocean-pipeline-scm-api'         => {},
+    'blueocean-rest'                     => {},
+    'blueocean-rest-impl'                => {},
+    'blueocean-web'                      => {},
+    'bouncycastle-api'                   => {},
+    'brakeman'                           => {},
+    'branch-api'                         => {},
+    'build-flow-plugin'                  => {},
```
Comment (ekohl) [inline puppet/modules/jenkins_master/manifests/init.pp]: Latest gives a 404 so can be dropped.

**Comment by tbrisker** [comment]
I've updated the unmaintained plugins to specify a version so we can first get this for the rest and then look into removing them separately to make sure we don't break something.

**Pull request merged**